src/package/packageConvert.ts

```diff
@@ -136,29 +145,72 @@ export async function convertPackage(
  * @private
  */
 export async function createPackageVersionCreateRequest(
-  context: { installationkey?: string; buildinstance?: string },
-  packageId: string
+  context: { installationkey?: string; definitionfile?: string; buildinstance?: string },
+  packageId: string,
+  apiVersion: string
 ): Promise<PackagingSObjects.Package2VersionCreateRequest> {
   const uniqueId = uniqid({ template: `${packageId}-%s` });
   const packageVersTmpRoot = path.join(os.tmpdir(), uniqueId);
+  const packageVersMetadataFolder = path.join(packageVersTmpRoot, 'md-files');
   const packageVersBlobDirectory = path.join(packageVersTmpRoot, 'package-version-info');
+  const settingsZipFile = path.join(packageVersBlobDirectory, 'settings.zip');
+  const metadataZipFile = path.join(packageVersBlobDirectory, 'package.zip');
   const packageVersBlobZipFile = path.join(packageVersTmpRoot, consts.PACKAGE_VERSION_INFO_FILE_ZIP);
 
   const packageDescriptorJson = {
     id: packageId,
   };
 
+  const settingsGenerator = new SettingsGenerator({ asDirectory: true });
+  const definitionFile = context.definitionfile;
+
+  try {
+    if (definitionFile) {
+      const definitionFilePayload = await fs.promises.readFile(definitionFile, 'utf8');
+      const definitionFileJson = JSON.parse(definitionFilePayload) as ScratchOrgInfo;
```

Comment:
@smotwani11 how do you feel about placing a try/catch around the read and parse that throws the `errorReadingDefintionFile` error and then removing the block try/catch?

Reply:
I will resolve this.